src/screens/send/components/address-book/address-book.tsx

```diff
@@ -0,0 +1,211 @@
+import { confirm } from '../../../../core/utils/dialog';
+import { IAccountState } from '../../../../redux/wallets/state';
+
+export interface IReduxProps {
+    contacts: IContactsState[];
+    deleteContact: typeof deleteContact;
+    updateContactName: typeof updateContactName;
+}
+
+export interface IExternalProps {
+    blockchain: Blockchain;
+    accounts: IAccountState[];
+    onAccountSelection: (account: IAccountState) => any;
+}
+
+interface IState {
+    openedSwipeIndex: string;
+}
+
+export const mapStateToProps = (state: IReduxState, ownprops: IExternalProps) => {
+    return {
+        contacts: selectContacts(state, ownprops.blockchain)
+    };
+};
+
+const mapDispatchToProps = {
+    deleteContact,
+    updateContactName
+};
+
+export class AddressBookComponent extends React.Component<
+    IReduxProps & IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>,
+    IState
+> {
+    public walletSwipableRef: any[] = new Array();
+
+    constructor(
+        props: IReduxProps & IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>
+    ) {
+        super(props);
+        this.state = {
+            openedSwipeIndex: null
+        };
+    }
+
+    public closeCurrentOpenedSwipable() {
+        this.walletSwipableRef[this.state.openedSwipeIndex] &&
+            this.walletSwipableRef[this.state.openedSwipeIndex].close();
+    }
+
+    public onPressUpdate(contact: IContactState) {
+        const title = translate('Send.alertEditTitle');
+        const message = translate('Send.alertEditDescription');
+        const buttons = [
+            {
+                text: translate('App.labels.cancel'),
+                onPress: () => {
+                    /* console.log('Cancel Pressed')*/
+                },
+                type: 'cancel'
+            },
+            {
+                text: translate('App.labels.save'),
+                onPress: (inputValue: string) => {
+                    if (inputValue !== '') {
+                        const data: IContactState = {
+                            address: contact.address,
+                            blockchain: contact.blockchain,
+                            name: inputValue
+                        };
+                        this.closeCurrentOpenedSwipable();
+                        this.props.updateContactName(data);
+                    }
+                },
+                type: 'default'
+            }
+        ];
+        const type = 'plain-text';
+
+        Alert.prompt(title, message, buttons, type);
+    }
+
+    public renderLeftActions = (contact: IContactState) => {
+        const styles = this.props.styles;
+        return (
+            <View style={styles.leftActionsContainer}>
+                <TouchableOpacity
+                    style={styles.action}
+                    onPress={() => {
+                        confirm(translate('Send.deleteContact'), '')
```

Comment:
you can move the confirm in `deleteContact` action creator, so from here you only call `this.props.deleteContact(contact);`. This way this would be more reusable.